### PR TITLE
Revert signer change and add shard info in tx hash for eip155

### PR DESCRIPTION
**Pull request by rlan35**
We shouldn't change the signer without adding correct forking logic.

### CODE REVIEW

**Comment by coolcottontail** [comment]
just FYI... once this is reverted. All transactions will be signed using old signer as blockNum is small.  The wallet won't be able to submit new transactions.

File: core/types/transaction_signing.go [hunk 1]
```diff
@@ -42,7 +42,14 @@ type sigCache struct {
 // MakeSigner returns a Signer based on the given chain config and block number.
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int) Signer {
 	var signer Signer
-	signer = NewEIP155Signer(config.ChainID)
+	switch {
+	case config.IsEIP155(blockNumber):
+		signer = NewEIP155Signer(config.ChainID)
+	case config.IsHomestead(blockNumber):
```
Comment (LeoHChen) [inline core/types/transaction_signing.go]: why do we need the homestead signer?

**Comment by rlan35** [comment]
> just FYI... once this is reverted. All transactions will be signed using old signer as blockNum is small. The wallet won't be able to submit new transactions.

Good catch. also reverted wallet signing part.

**Comment by rlan35** [comment]
> What about the wallet signer part? It uses nil as the chain config

also reverted.

**Pull request merged**